### ISSUE TITLE
Feature requests go here!

### ISSUE THREAD

**Issue by MizaGBF** [MizaGBF/GBFAL#3]
If there is some sort of feature or change that you would wish to see implemented, you can post it here.  

Here's a detailed list of what I might not, or won't, bother with:  
- Items icons  
  
You can easily find those in-game and it wouldn't add much value. The only interesting point would be for cut-content items but there are only a few of them.

- Anything fate episode related

The first hurdle would be to have a reliable way to track which fate are added to the game without having to crawl over all of them.
Second, considering the amount of episodes in-game, it would be quite time and bandwidth consuming.

- Event voice files

I actually got a way to get those but it would make the events even harder to maintain over time and I'm not sure Cygames would like it either.

- BGM and boss voice files

They simply don't have a reliable naming scheme which we can exploit to find them.

- Skill sprite sheets and SFX files

They wouldn't add much values, although I'm not considering those to be off the table forever.

tldr: Anything sound or fate episode related is likely off the table.

**Comment by jiaoxiaohang** [comment]
![image](https://github.com/MizaGBF/GBFAL/assets/69664925/00141618-4c60-424e-857a-4639814a01e4)
Hello!
Can update records be displayed for longer?
Thank you!

**Comment by jiaoxiaohang** [comment]
![image](https://github.com/MizaGBF/GBFAL/assets/69664925/1a1ce315-1a96-4b18-9982-87ece819e537)
Hello!
Can you add illustrations of skycompass? The website in the picture will stop updating in 2021.
Thank you!

**Comment by MizaGBF** [comment]
> ![image](https://private-user-images.githubusercontent.com/69664925/317976203-00141618-4c60-424e-857a-4639814a01e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MDM1OTksIm5iZiI6MTcxMTcwMzI5OSwicGF0aCI6Ii82OTY2NDkyNS8zMTc5NzYyMDMtMDAxNDE2MTgtNGM2MC00MjRlLTg1N2EtNDYzOTgxNGEwMWU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI5VDA5MDgxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViMWNiOWZiYWQ3NjAxZWE2NjBjNjQ5NmNlNzUxMmQyZjEyNGYyZjE1MzBhYzRhNGU5Njg2ZjZhNGFmNTRiZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g3oTpo02NILBfDnVFDDKNr0W46e-BknRz4szj2K0nGo) Hello! Can update records be displayed for longer? Thank you!

No, it's supposed to show only the recently updated elements, I played around with the limit a lot in the past and I think it's currently fine, I don't want to make the loading even heavier.

> ![image](https://private-user-images.githubusercontent.com/69664925/317976523-1a1ce315-1a96-4b18-9982-87ece819e537.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MDM1OTksIm5iZiI6MTcxMTcwMzI5OSwicGF0aCI6Ii82OTY2NDkyNS8zMTc5NzY1MjMtMWExY2UzMTUtMWE5Ni00YjE4LTk5ODItODdlY2U4MTllNTM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI5VDA5MDgxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjZjEzYzU5MzM0ZjhmNTZkZDUyMTNkMWE4YWRlMDM2NWI3ZmYyMjNhYTBhMzEzYjA4ZDZhMGJhYTMxNmIzYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1y20cydTwf4_4TQvqtcStypYDseIXXmJw1Hbxa5Gd6o) Hello! Can you add illustrations of skycompass? The website in the picture will stop updating in 2021. Thank you!

I'm pretty sure the wiki got them all [here](https://gbf.wiki/Promotional_Art).
Plus, I have no idea how to retrieve those from the sky compass.

**Comment by jiaoxiaohang** [comment]
> > ![图像](https://private-user-images.githubusercontent.com/69664925/317976203-00141618-4c60-424e-857a-4639814a01e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MDM1OTksIm5iZiI6MTcxMTcwMzI5OSwicGF0aCI6Ii82OTY2NDkyNS8zMTc5NzYyMDMtMDAxNDE2MTgtNGM2MC00MjRlLTg1N2EtNDYzOTgxNGEwMWU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI5VDA5MDgxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViMWNiOWZiYWQ3NjAxZWE2NjBjNjQ5NmNlNzUxMmQyZjEyNGYyZjE1MzBhYzRhNGU5Njg2ZjZhNGFmNTRiZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g3oTpo02NILBfDnVFDDKNr0W46e-BknRz4szj2K0nGo)你好！更新记录可以显示更长时间吗？谢谢！
> 
> 不，它应该只显示最近更新的元素，我过去经常玩过限制，我认为目前很好，我不想让负载更重。
> 
> > ![图像](https://private-user-images.githubusercontent.com/69664925/317976523-1a1ce315-1a96-4b18-9982-87ece819e537.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MDM1OTksIm5iZiI6MTcxMTcwMzI5OSwicGF0aCI6Ii82OTY2NDkyNS8zMTc5NzY1MjMtMWExY2UzMTUtMWE5Ni00YjE4LTk5ODItODdlY2U4MTllNTM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI5VDA5MDgxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjZjEzYzU5MzM0ZjhmNTZkZDUyMTNkMWE4YWRlMDM2NWI3ZmYyMjNhYTBhMzEzYjA4ZDZhMGJhYTMxNmIzYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1y20cydTwf4_4TQvqtcStypYDseIXXmJw1Hbxa5Gd6o)你好！你能添加天罗盘的插图吗？图中网站将于2021年停止更新。谢谢！
> 
> 我很确定 wiki 把它们都[放在这里](https://gbf.wiki/Promotional_Art)。另外，我不知道如何从天空罗盘中检索它们。

Thank you, the illustrations here are really comprehensive

**Comment by M1hono** [comment]
Hi! Thanks for your excellent job! Have some issue for your help.
Is this possible to connect NPC's ID with characters' name? Now it's a bit messy for researching and finding.
And also, I'm creating a bot app for searching characters, is there any API provided for the developers?

**Comment by MizaGBF** [comment]
> Is this possible to connect NPC's ID with characters' name?
  
Not that I know of.  
> And also, I'm creating a bot app for searching characters, is there any API provided for the developers?
  
No, it's a static webpage. I think you would have better luck using the gbf.wiki API, as it's what I use. Then, if you somehow want to link to my page, just append the ID of the character/element to `https://mizagbf.github.io/GBFAL/?id=`.
Example: [https://mizagbf.github.io/GBFAL/?id=3040036000](https://mizagbf.github.io/GBFAL/?id=3040036000)
If you need specific data from GBFAL itself, simply request the data.json file from the repo (not from the website, this way you're certain to have the last one and it won't cause any bandwidth problem): `https://raw.githubusercontent.com/MizaGBF/GBFAL/main/json/data.json`

**Comment by M1hono** [comment]
> > Is this possible to connect NPC's ID with characters' name?
> 
> Not that I know of.
> 
> > And also, I'm creating a bot app for searching characters, is there any API provided for the developers?
> 
> No, it's a static webpage. I think you would have better luck using the gbf.wiki API, as it's what I use. Then, if you somehow want to link to my page, just append the ID of the character/element to `https://mizagbf.github.io/GBFAL/?id=`. Example: https://mizagbf.github.io/GBFAL/?id=3040036000 If you need specific data from GBFAL itself, simply request the data.json file from the repo (not from the website, this way you're certain to have the last one and it won't cause any bandwidth problem): `https://raw.githubusercontent.com/MizaGBF/GBFAL/main/json/data.json`

That's exactly what I'm looking for! Many thanks.

**Comment by MizaGBF** [comment]
I added an in-depth breakdown of GBFAL data file [here](https://github.com/MizaGBF/GBFAL/blob/main/docs.md), as I expect I might get more questions of this kind, in the future, and I'd rather people read this instead. I hope that I didn't make mistakes.  
(It might prove useful to myself too).

**Comment by M1hono** [comment]
Is that possible to track story files?

**Comment by MizaGBF** [comment]
Story files are already tracked, unless you meant something else.

**Comment by M1hono** [comment]
I mean the Story files that contain all of lore. May I see where they are saved in FAL?

**Comment by MizaGBF** [comment]
You mean the scene texts/scripts?
Nope, I don't intend to go that route, for many reasons:
1. It's not accessible without an account. Plus, stuff like fate episodes also require the character unlocked and events require to have played it, and I certainly don't have/haven't done all of them.  
2. And automating the retrieval process wouldn't be possible without breaking the game ToS a few times.  
3. Plus, I would have to host the text directly (impossible to hot link it like I do currently for the assets), opening to the potential issue of rehosting this copyrighted content.  
4. The story is already archived [here](https://calico.nekoweb.org/gbf/), while the wiki already got most of everything else.  
5. Finally, the Asset Lookup is meant to be for graphic assets only. Sound files being a small exception I made.  
  
I did have the idea of making some sort of Scene Player (akin to the Animation Player), where people could just paste a scene JSON script and have it play. But I haven't tested how feasible it would be and I'm pretty sure it would require a proxy like the Animation Player to play the few animations. And, as those are just being fun little projects, I don't intend to invest money into them to make it viable either.

**Comment by M1hono** [comment]
Augh I see but the problem is that sometimes I missed side stories which is kind pain.
You made your points and I understood it. Wiki stuff is good enought though.
